Orchestrator cancellation does not abort in-flight HTTP requests

## Bug

When a user cancels an orchestration:
1. The forward task breaks its event loop and drops the channel receiver
2. The ChatModelWorker continues running — its in-flight HTTP request is never aborted
3. When the request eventually completes (or times out at 600s), the worker tries to send on the closed channel:

```
[ERROR] Worker error: Failed to send error event: channel closed
```

## Root Cause

Three design gaps in the cancellation flow:

1. **worker.cancel() is never called** — ChatModelWorker has a cancel() method and an Arc<Mutex<bool>> cancelled flag with checks at multiple points, but the orchestrator service never invokes it
2. **No HTTP abort token** — reqwest requests have no cancellation mechanism; they block until response or the 600s timeout
3. **Channel drops before worker finishes** — the forward task exits immediately on cancel, dropping the receiver while the worker is still alive

## Fix

1. Call worker.cancel() when cancellation signal is received
2. Wrap HTTP send().await in tokio::select! with the cancellation signal
3. Downgrade channel closed errors after cancellation to debug-level logs

## Impact

- Orphaned HTTP requests waste resources (up to 10 min per cancelled request)
- Noisy error logs on every cancellation
- Gateway 504 timeouts logged as errors when they are expected after cancel

--- a/src-tauri/src/orchestrator/chat_model_worker.rs
+++ b/src-tauri/src/orchestrator/chat_model_worker.rs
@@ -1072,13 +1072,22 @@ impl Worker for ChatModelWorker {
             if !response.status().is_success() {
                 let status = response.status();
                 let body_text = response.text().await.unwrap_or_default();
+                // Check cancellation before logging — a 504 after cancel is expected
+                if *self.cancelled.lock().await {
+                    log::debug!("[ChatModelWorker] HTTP {} after cancellation (expected)", status);
+                    return Ok(());
+                }
                 log::error!("[ChatModelWorker] HTTP {} from Gateway", status);
-                event_tx
+                if let Err(e) = event_tx
                     .send(WorkerEvent::Error {
                         message: format!("Gateway returned HTTP {}", status),
                     })
                     .await
-                    .map_err(|e| format!("Failed to send error event: {}", e))?;
+                {
+                    // Channel closed — likely cancelled while sending
+                    log::debug!("[ChatModelWorker] Channel closed, cannot send error: {}", e);
+                    return Ok(());
+                }
                 return Err(format!(
                     "Gateway returned HTTP {}: {}",
                     status,
@@ -1111,7 +1120,7 @@ impl Worker for ChatModelWorker {
                         final_content.len(),
                         total
                     );
-                    event_tx
+                    if let Err(e) = event_tx
                         .send(WorkerEvent::Complete {
                             final_content,
                             thinking,
@@ -1119,7 +1128,10 @@ impl Worker for ChatModelWorker {
                             rlm_steps: None,
                         })
                         .await
-                        .map_err(|e| format!("Failed to send Complete event: {}", e))?;
+                    {
+                        log::debug!("[ChatModelWorker] Channel closed, cannot send Complete: {}", e);
+                        return Ok(());
+                    }
                     log::info!("[ChatModelWorker] Execution complete, breaking from tool loop");
                     break;
                 }

--- a/src-tauri/src/orchestrator/service.rs
+++ b/src-tauri/src/orchestrator/service.rs
@@ -291,6 +291,7 @@ async fn execute_single_task(
         // Create channel and spawn worker
         let (event_tx, mut event_rx) = mpsc::channel::<WorkerEvent>(256);
         let worker = create_worker(&routing, app, &capabilities);
+        let worker_for_cancel = Arc::clone(&worker);
         let worker_prompt = subtask.prompt.clone();
         let worker_routing = routing.clone();
         let worker_app = app.clone();
@@ -315,11 +316,13 @@ async fn execute_single_task(
         let app_for_events = app.clone();
         let cancel_rx_clone = cancel_rx.clone();
 
-        // Collect all events, intercepting errors for reroute analysis
+        // Collect all events, intercepting errors for reroute analysis.
+        // Returns (was_cancelled, captured_error).
         let mut reroutable_error: Option<String> = None;
         let forward_handle = tokio::spawn(async move {
             let mut taken_rx = cancel_rx_clone.lock().await.take();
             let mut captured_error: Option<String> = None;
+            let mut cancelled = false;
             loop {
                 if let Some(ref mut rx) = taken_rx {
                     tokio::select! {
@@ -345,6 +348,7 @@ async fn execute_single_task(
                         }
                         _ = rx => {
                             log::info!("[Orchestrator] Cancellation received for conversation {}", conv_id);
+                            cancelled = true;
                             break;
                         }
                     }
@@ -371,12 +375,22 @@ async fn execute_single_task(
                     }
                 }
             }
-            captured_error
+            (cancelled, captured_error)
         });
 
         let forward_result = forward_handle.await;
-        if let Ok(Some(error_msg)) = &forward_result {
+        let was_cancelled = forward_result.as_ref().map(|(c, _)| *c).unwrap_or(false);
+        if let Ok((_, Some(ref error_msg))) = forward_result {
             reroutable_error = Some(error_msg.clone());
+        }
+
+        // If the forward loop exited due to cancellation, signal the worker
+        // to stop and abort its task so in-flight HTTP requests don't linger.
+        if was_cancelled {
+            log::info!("[Orchestrator] Cancelling worker for conversation {}", conversation_id);
+            let _ = worker_for_cancel.cancel().await;
+            worker_handle.abort();
+            break;
         }
 
         match worker_handle.await {

